File: Atividade IV/problema17.hs
mdc :: Int -> Int -> Int
mdc _x 0 = _x
mdc x y =
	mdc y (mod x y)

main = do
	as <- getLine
	bs <- getLine
	let a = (read as :: Int)
	let b = (read bs :: Int)
	if a > b then
		print(mdc a b)
	else
		print(mdc b a)

File: PROLOG/Atividade XV/problema17.pl
mdc(X,Y,Resultado) :-
    Resto is X mod Y,
    (Resto == 0 -> Resultado is Y; mdc(Y,Resto,Resultado)).

mmc(X,Y,Resultado) :-
    mdc(X,Y,R1),
    Resultado is (X*Y)/R1.
% ?- mmc(45,32,R).
% R = 1440.